### PR TITLE
Refactor sap system selectors

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Refactor sap systems/databases selectors to use `createSelector`. This will enable to update SapSystemsOverview and DatabasesOverview views to make storybook addition easy.

Besides that, I have moved the functions to a new `sapSystems` file, used lodash in some operations and removed the camelCase transformation that is only used in these selectors.

## How was this tested?

Tests added after refactoring

### CODE REVIEW

File: assets/js/state/selectors/cluster.js [hunk 1]
```diff
@@ -24,6 +24,11 @@ export const getClusterName = (clusterID) => (state) => {
   return get(cluster, 'name', '');
 };
 
+export const getClusterByHost = (state, hostID) => {
```
Comment (arbulu89) [inline assets/js/state/selectors/cluster.js:27]: Moved from `index`. Now it uses lodash find function

File: assets/js/state/selectors/sapSystems.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  state.sapSystemsList.sapSystems.find((sapSystem) => id === sapSystem.id);
+
+export const getDatabase = (id) => (state) =>
+  state.databasesList.databases.find((database) => id === database.id);
+
+export const getEnrichedSapSystemDetails = createSelector(
+  [
+    (id, state) => getSapSystem(id)(state),
+    (id, state) =>
+      enrichInstances(
```
Comment (arbulu89) [inline assets/js/state/selectors/sapSystems.js:47]: I'm not sure if adding this operation here is a good idea, or it should better go in the body of the function

File: assets/js/state/selectors/sapSystem.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  (enrichedInstances) => enrichedInstances
+);
+
+export const getSapSystem = (id) => (state) =>
+  state.sapSystemsList.sapSystems.find((sapSystem) => id === sapSystem.id);
+
+export const getDatabase = (id) => (state) =>
+  state.databasesList.databases.find((database) => id === database.id);
+
+export const getEnrichedSapSystemDetails = createSelector(
```
Comment (arbulu89) [inline assets/js/state/selectors/sapSystem.js:43]: In essence, the next `getEnrichedSapSystemDetails` and `getEnrichedDatabaseDetails` functions have been changes in the whole PR.

File: assets/js/state/selectors/sapSystem.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  [
+    (id, state) => getSapSystem(id)(state),
+    (id, state) =>
```
Comment (dottorblaster) [inline assets/js/state/selectors/sapSystem.js:46]: Can you switch this to `(state, id)` form? Usually `state` comes as the first parameter when you use `createSelector`

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just a thing then we can merge. Awesome work, thank you so much!

File: assets/js/state/selectors/cluster.js (hunk 1 above)
Comment (dottorblaster) [reply assets/js/state/selectors/cluster.js:27]: YEEEEEEEAH

File: assets/js/state/selectors/sapSystem.js (hunk 4 above)
Comment (arbulu89) [reply assets/js/state/selectors/sapSystem.js:46]: I can and I will!
Thank you for the tip

Comment (arbulu89) [reply assets/js/state/selectors/sapSystem.js]: Done

**Review by dottorblaster — APPROVED** [review]
LGTM

**Review by rtorrero — APPROVED** [review]
Thanks @arbulu89, this looks much clearer now

**Pull request merged**